roxie/ccd/ccdqueue.cpp

```diff
@@ -610,38 +608,40 @@ void SlaveContextLogger::flush()
 
 //=================================================================================
 
-unsigned getIbytiDelay(unsigned channel, const RoxiePacketHeader &header)
+bool isSubChannelAlive(unsigned channel, unsigned subChannel)
 {
-    // MORE - adjust delay according to whether it's a retry, whether it was a broadcast etc
-    CriticalBlock b(ibytiCrit);
-    return IBYTIDelays[channel];
+    unsigned mask = SUBCHANNEL_MASK << (SUBCHANNEL_BITS * subChannel);
+    return (liveSubChannels[channel] & mask) == mask;
 }
 
-void resetIbytiDelay(unsigned channel)
+unsigned getIbytiDelay(unsigned channel, unsigned primarySubChannel)
 {
-    unsigned prevVal;
-    {
-        CriticalBlock b(ibytiCrit);
-        prevVal = IBYTIDelays[channel];
-        IBYTIDelays[channel] = initIbytiDelay;
-    }
-    if (traceLevel > 8 && prevVal != initIbytiDelay)
-        DBGLOG("Reset IBYTI delay value for channel %u from %u to %u", channel, prevVal, initIbytiDelay);
+    // MORE - adjust delay according to whether it's a retry, whether it was a broadcast etc
+    unsigned mySubChannel = subChannels[channel] - 1;
+    unsigned subChannel = primarySubChannel;
+    unsigned delay = 0;
+    while (subChannel != mySubChannel)
```

Comment:
Do you want to loop over all others ?

Reply:
Nit sure I understand the question... the intent is that every time there is another slave that should be expected to handle this before me, and that slave is alive, I should add one to the delay factor.